numpy/core/src/_simd/_simd.dispatch.c.src

```diff
@@ -3,12 +3,6 @@
 #include "_simd_inc.h"
 
 #if NPY_SIMD
-#include "_simd_inc_data.h"
-#include "_simd_inc_convert.h"
-#include "_simd_inc_vector.h"
-#include "_simd_inc_arg.h"
-#include "_simd_inc_easyintrin.h"
-
```

Comment:
They can't be used independentely and also needs a better guard(the main header guard) on the top of each one of them,
this pattern is not strange and already been used by many API, e.g. `immintrin.h`.

Reply:
`immintrin.h` is a convenient file that lumps all the includes together for the user, _and does nothing else_. That's not what you're doing here.

As mattip points out in https://github.com/numpy/numpy/pull/16782#discussion_r457450839, these should really be `C` files, not headers at all. Is that possible?

Reply:
> immintrin.h is a convenient file that lumps all the includes together for the user, and does nothing else. That's not what you're doing here.

That's exectly what I'm doing here, the only thing is defferent here that I tend to use `static` instead to avoid giving an `inline` hint to the compiler to reduce the binary size, maybe its better to use `inline` with `__attribute__((noinline))` or just add new policy to `CCompilerOpt` for optimization for code size(`-Os`)

> As mattip points out in #16782 (comment), these should really be C files, not headers at all. Is that possible?

create a new dispatch-able source gathering all functions will requires the use of macros `NPY_CPU_DISPATCH_DECLARE` and  `NPY_CPU_DISPATCH_CURFX` to avoid duplicating symbols,
therfore headers are used instead.